drivers/infiniband/hw/ntrdma/ntrdma_vbell.c

```diff
@@ -211,8 +212,11 @@ static void ntrdma_dev_vbell_work(struct ntrdma_dev *dev, int vec)
 		ntrdma_vbell_head_fire(head, vbell_val);
 	}
 
-	if (ntc_clear_signal(dev->ntc, vec))
-		tasklet_schedule(&dev->vbell.work[vec]);
+	if (ntc_clear_signal(dev->ntc)) {
+		if (*counter == dev->ntc->peer_irq_num)
+			*counter = 0;
+		tasklet_schedule(&dev->vbell.work[*counter++]);
```

Comment:
if we cleared this irq (affinity to #cpu) then reschedule so we will not lose irq